### ISSUE TITLE
Update: Multiple occurances - p. 218 - Define API

### ISSUE THREAD

**Issue by wesbland** [mpi-forum/mpi-issues#195]
## Problem

API is used without definition, and only used occasionally.  A consistent term should be used for the MPI specification, and used consistently.

### Suggested Fix

Add to chapter 2?

**Comment by wesbland** [comment]
Voting:
* Editor - 3
* Chapter committee - 2
* Full proposal - 2

Personally, I think if we're going to add a new term, that's probably a full proposal. Let's discuss this in the virtual meeting.

**Comment by dholmes-epcc-ed-ac-uk** [comment]
IMHO, this ticket should probably be change "API" to "procedure" or "procedures" throughout the document, rather than add "API" as a new term.
There are only 18 instances of "API" but changing them will create merge conflicts with, e.g. sessions.
This should be a full proposal because it is not a mechanical replacement-for-synonym exercise.

**Comment by wesbland** [comment]
That makes a lot of sense to me and I agree that if we go that route, it's a full proposal.

**Comment by devreal** [comment]
A word of caution about replacing `API` with the term procedure: an API is more than just procedures and includes constants and exposed types. MPI is an API and is more than just a set of procedures.

I only find 6 mentions of API in the 3.1 document and half of them is in this paragraph on page 373:

> Complex interaction of an MPI application with its runtime environment should be done through an environment-specific API. An example of such an API would be the PVM task and machine management routines — pvm_addhosts, pvm_config, pvm_tasks, etc., possibly modified to return an MPI (group, rank) when possible. A Condor or PBS API would be another possibility.

(Side note: that's the only mention of PBS and Condor, without references.)

Then on page 399:

> It is the responsibility of the application to create the socket using standard socket API calls.


and on page 568:


> Except where otherwise noted, all conventions and principles governing the C bindings of the MPI API also apply to the MPI tool information interface, which is available by including the mpi.h header file.

In both cases, replacing API with "procedures" would be inaccurate.

The last instance I find is in the changelog on page 804:

> The read access restriction on the send buffer for blocking, non blocking and collective API has been lifted.

Here, replacing API with procedure is probably fine although I personally prefer API to remain there.

Overall, adding a definition of API (maybe upon its first mention) is a simple fix and does not require changing existing text.

**Comment by dholmes-epcc-ed-ac-uk** [comment]
I agree with @devreal that MPI procedure(s) is not appropriate for some instances of API.

I find 13 Latex lines that contain "API" in a more recent version of MPI.

Persistent collectives added quite a few new uses of API into the Collectives chapter and one into the changelog, most/all of which should not be API.

The one usage in the Tools chapter seems bad - the MPI API includes the MPI tool information interface.

There are two changes/tickets here:
* define API for the correct usages of it
* change API to something else for the incorrect usages of it

> Dan-Holmes-MBP:mpi-standard dholmes$ grep -Rn API --include=*tex .
./chap-changes/changes.tex:1231:The read access restriction on the send buffer for blocking, non blocking and collective API has been lifted. It is permitted to access for read the send buffer while the operation is in progress. 
>
> ./chap-tools/mpit.tex:70:bindings of the \MPI/ API also apply to the \MPI/ tool information
>
> ./chap-inquiry/inquiry.tex:538:from completing further API calls successfully and those functions will continue
>
> ./chap-dynamic/dynamic-2.tex:210:be done through an environment-specific API.
./chap-dynamic/dynamic-2.tex:211:An example of such an API would be the PVM task and machine management
./chap-dynamic/dynamic-2.tex:214:A Condor or PBS API would be another possibility. 
./chap-dynamic/dynamic-2.tex:1943:of the application to create the socket using standard socket API
./chap-dynamic/dynamic-2.tex:1993:% LocalWords:  SMP API pvm addhosts config SPMD MPMD argv maxprocs comm int ck
>
> ./chap-coll/coll.tex:3501:Nonblocking collective requests created using the APIs described in this section are not persistent. However, persistent collective requests can be created using persistent collective operations described in Sections~\ref{sec:coll-persistent} and~\ref{sec:sparsecoll-pnbc}.
./chap-coll/coll.tex:4273:a corresponding persistent collective operation with the analogous API
./chap-coll/coll.tex:4276:The collective persistent API signatures include an \const{MPI\_INFO} object in order to support
./chap-coll/coll.tex:4289:%The \mpiarg{info} arguments are added to the API to support non-standard optimization hints, and to allow future
>
> ./instr.tex:893:used to extract the MPI C API declarations.

**Comment by wesbland** [comment]
At this point, I assume this isn't going to happen for MPI 4.1. I'm going to move it to MPI 5.0 and let @puribangalore bring it back if he wants to do so.
